src/Spago/CLI.hs

```diff
@@ -162,6 +166,8 @@ parser = do
     buildOptions  = BuildOptions <$> watch <*> clearScreen <*> allowIgnored <*> sourcePaths <*> srcMapFlag <*> noInstall
                     <*> pursArgs <*> depsOnly <*> beforeCommands <*> thenCommands <*> elseCommands
 
+    versionOpt  = Opts.flag' DoShowVersion (Opts.long "version" <> Opts.help "Show spago version")
```

Comment:
Is there a reason why we don't have to pass `NoShowVersion` here?

Reply:
Yes. [`Opts.flag'`](https://hackage.haskell.org/package/optparse-applicative-0.15.1.0/docs/Options-Applicative.html#v:flag-39-) creates kind of a "partial" parser. That is, it does not have a default value. This is intended. We want this parser to fail if it does _not_ get a `--version` flag, rather than defaulting to `NoShowVersion`. We rely on the parser failure to show the help message. This is also the reason why we just bail out of the `handleDefault` function with `pure ()`. Because (as far as I could see) there is no way to get and propagate the help message to `handleDefault` if we successfully parse the command. So we rely on parser failure instead.

Reply:
Yep, I figured something like this was happening - could you add this explanation as a comment above this line?